File: task1.py
def max_two_two_digit_sum_verbose(s):
    n = len(s)
    subRet = float('-inf')

    #subString 2개 조합 int형 변환
    twoDigitNumbers = []
    for i in range(n - 1):
        two = s[i:i+2]
        twoDigitNumbers.append(int(two))

    # 겹치는 값 중 최대값 만들기
    fragments = [0] * len(twoDigitNumbers)
    for i in range(len(twoDigitNumbers)):
        if twoDigitNumbers[i] > subRet:
            subRet = twoDigitNumbers[i]
        fragments[i] = subRet

    ans = float('-inf')

    # [0,2], [1,3] 겹치지않는 범위에서 최대값 체크
    for i in range(2, len(twoDigitNumbers)):
        left = fragments[i - 2]
        right = twoDigitNumbers[i]

        ans = max(ans, left + right)

    return ans
print(max_two_two_digit_sum_verbose('9999999999999999999999999999999999999999999999999999999999999999999999999999999999'))

File: task1.java
import java.util.Scanner;

public class task1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String n = sc.nextLine();

        //32481 81 + 32
        Long ret = Long.MIN_VALUE;
        Long[] data = new Long[n.length()-1];
        for (int i = 0; i < n.length()-1; i++) {
            data[i] = Long.parseLong(n.substring(i,i+2));
        }

        Long overLappingValue = data[0];
        //간격중에 큰 값을 max값으로 선택한다.
        Long[] sub_ret = new Long[data.length];
        for (int i = 0; i < data.length-1; i++) {
            //만약 ret보다 data[i]가크면 업데이트
            if(overLappingValue < data[i]) {
                overLappingValue = data[i];
            }
            sub_ret[i] = overLappingValue;
        }

        for (int i = 0; i < data.length; i++) {
            System.out.println(data[i]);
        }

        long ret2 = Long.MIN_VALUE;
        for(int i=2; i<sub_ret.length; i++) {
            long left = sub_ret[i-2];
            long right = data[i];

            ret2 = Math.max(ret2, left+right);
        }

        System.out.println(ret2);


    }
}